modules/rostests/apitests/shell32/ShellExecuteW.cpp

```diff
@@ -0,0 +1,185 @@
+                     wine_dbgstr_w(Name), GetLastError());
+
+    if (hInstance) 
+    {
+        LPCWSTR lpWinClass = L"CabinetWClass";
+        LPCWSTR lpWinTitleWindows = L"C:\\Windows";
+        LPCWSTR lpWinTitleReactOS = L"C:\\ReactOS";
+
+        Sleep(1000);
+        if (IsReactOS)
+            hWnd = FindWindowW(lpWinClass, lpWinTitleReactOS);
+        else
+            hWnd = FindWindowW(lpWinClass, lpWinTitleWindows);
+        PostMessage(hWnd, WM_SYSCOMMAND, 0xF060, 0);  // Terminate Window
+    }
+// End of test #3 - Open C:\Windows
+
+
+//  ShellExecuteW(handle, "open", <fully_qualified_path_to_folder>, NULL, NULL, SW_SHOWNORMAL);
+
+    if(IsReactOS)
+        hInstance = ShellExecuteW(NULL, L"open", L"C:\\ReactOS\\system32", NULL,
+                                  NULL, SW_SHOWNORMAL);
+    else
+        hInstance = ShellExecuteW(NULL, L"open", L"C:\\Windows\\system32", NULL,
+                                  NULL, SW_SHOWNORMAL);
+
+    retval = (UINT_PTR) hInstance;
+
+    printf("Return Value for C:\\...\\system32 is '%d'.\n", retval);
+
+    ok(retval > 31, "ShellExecuteEx lpFile %s failed. Error: %lu\n",
+                     wine_dbgstr_w(Name), GetLastError());
+
+    if (hInstance)
+    {
+        LPCWSTR lpWinClass = L"CabinetWClass";
+        LPCWSTR lpWinTitle = L"C:\\Windows\\system32";
+        Sleep(1000);
+        hWnd = FindWindowW(lpWinClass, lpWinTitle);
+        PostMessage(hWnd, WM_SYSCOMMAND, 0xF060, 0);  // Terminate Window
+    }
+// End of test #4 - Open C:\Windows\system32
+
+
+//  ShellExecuteW(handle, "explore", <fully_qualified_path_to_folder>, NULL, NULL, SW_SHOWNORMAL);
+
+    hInstance = ShellExecuteW(NULL, L"explore", L"C:\\", NULL, NULL, SW_SHOWNORMAL);
+    retval = (UINT_PTR) hInstance;
+
+    printf("Return Value for explore c:\\ is '%d'.\n", retval);
+
+    ok(retval > 31, "ShellExecuteEx lpFile %s failed. Error: %lu\n",
+                     wine_dbgstr_w(Name), GetLastError());
+
+    if (hInstance)
+    {
+        LPCWSTR lpWinClass = L"ExploreWClass";
+        LPCWSTR lpWinTitle = L"C:\\";
+        Sleep(1000);
+        hWnd = FindWindowW(lpWinClass, lpWinTitle);
+        PostMessage(hWnd, WM_SYSCOMMAND, 0xF060, 0);  // Terminate Window
+    }
+// End of test #5 - explore C:
+
+
+//  ShellExecuteW(handle, "find", <fully_qualified_path_to_folder>, NULL, NULL, 0);
+
+    hInstance = ShellExecuteW(NULL, L"find", L"C:\\", NULL, NULL, SW_SHOWNORMAL);
+    retval = (UINT_PTR) hInstance;
+
+    printf("Return Value for find is '%d'.\n", retval);
+
+    ok(retval > 31, "ShellExecuteEx lpFile %s failed. Error: %lu\n",
+                     wine_dbgstr_w(Name), GetLastError());
+
+    if (hInstance)
+    {
+        LPCWSTR lpWinClass = L"CabinetWClass";
+        LPCWSTR lpWinTitle = L"Search Results";
+        Sleep(1000);
+        hWnd = FindWindowW(lpWinClass, lpWinTitle);
+        PostMessage(hWnd, WM_SYSCOMMAND, 0xF060, 0);  // Terminate Window
```

Comment:
0xF060 is SC_CLOSE

Reply:
This works to close the notepad window, but none of the other windows on ReactOS, but it works well in Windows Server 2003 to close all of the windows opened on the test. This probably deserves a JIRA entry itself.